File: crypto/vignere.js
// MODULE VIGNERE CIPHER

// Function trim: trimming whitespaces between message
function trim (msg) {
    var trimmed = "";
    for (var i = 0; i < msg.length; i++) {
        if (msg.charAt(i) != ' ') {
            trimmed += msg.charAt(i);
        }
    }

    return trimmed;
}

// Function encrypt: encrypting using standard Vignere Cipher
function encrypt (plainText, key) {
    var cipherText = "";
    for (var i = 0; i < plainText.length; i++) {
        var ord = (plainText.charCodeAt(i) - 65) + (key.charCodeAt(i % (key.length)) - 65);
        cipherText += ord.fromCharCode(0);

        // for each 5 characters, get spaced
        if (i % 5 == 0) {
            cipherText += " ";
        }
    }

    return cipherText;
}

// Function decrypt: decrypting using standard Vignere Cipher
function decrypt (cipherText, key) {
    var ctext = trim(cipherText);
    var plainText = "";
    for (var i = 0; i < ctext.length; i++) {
        var ord = (ctext.charCodeAt(i) - 65) - (key.charCodeAt(i % (key.length)) - 65);
        plainText += ord.fromCharCode(0);
    }

    return plainText;
}

module.exports.encrypt = encrypt;
module.exports.decrypt = decrypt;

File: classic-cipher/vignere.py
def encrypt(plaintext, key):
'''
For encrypting plaintext with Vignere Cipher
'''
    ciphertext = ""
    for i in range(len(plaintext)):
        c_val = chr(
            ord(plaintext[i]) - ord('A') + ord(key[i % len(key)])
        )
        ciphertext += c_val

        if i % 5 == 0 and i > 0:
            ciphertext += ' ' 

    return ciphertext


def decrypt(ciphertext, key):
'''
For decrypting plaintext with Vignere Cipher
'''
    plaintext = ""
    for i in range(len(ciphertext)):
        c_val = chr(
            ord(ciphertext[i]) - ord(key[i % len(key)]) + ord('A')
        )
        plaintext += c_val

    return plaintext
